walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,18 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
```

Comment:
recommend removing this instead of commenting, now that you have added code

Reply:
Sounds good! I had not removed it because I was not entirely sure what it was

Reply:
Usually stub methods or empty methods have them to serve as a reminder . There is also another important use of throwing such an exception - when the design of the class is such that the method is not supported. For example, let's take the `add` method in `AbstractList` - https://github.com/openjdk-mirror/jdk7u-jdk/blob/f4d80957e89a19a29bb9f9807d2a28351ed7f7df/src/share/classes/java/util/AbstractList.java#L148

AbstractList as the name goes is abstract, so adding something to it wouldn't make sense. Hence the method throws the exception. Now, classes inheriting `AbstractList`(like `ArrayList`) will either override the method to provide their own implementation or inherit the existing behavior (which is to throw the exception).  Since `ArrayList` supports addition of elements at a specific index, it has its own implementation - https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/util/ArrayList.java#L426